llvm/lib/Target/DirectX/DXIL.td

```diff
@@ -218,6 +218,9 @@ class DXILOpMapping<int opCode, DXILOpClass opClass, Intrinsic intrinsic, string
 // Concrete definition of DXIL Operation mapping to corresponding LLVM intrinsic
 def Sin  : DXILOpMapping<13, unary, int_sin,
                          "Returns sine(theta) for theta in radians.">;
+def Exp : DXILOpMapping<21, unary, int_exp,
+                         "Returns the base-e exponential of the x parameter."
```

Comment:
DXIL exp is 2^x instead of e^x.
For hlsl, there're exp and exp2(https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-exp2).
exp2 is the one mapped to DXIL exp directly.

Reply:
I'm confused by this answer opcode 21 is `exp` not` exp2` https://godbolt.org/z/MexG31jvv
exp docs says e^x  https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-exp

Reply:
ok I see whats going on. seems like we should make exp2 the intrinsic then do exp separately.